### ISSUE TITLE
CLIP-IQA+ pretrained weights

### ISSUE THREAD

**Issue by LorenzoAgnolucci** [chaofengc/IQA-PyTorch#137]
HI @chaofengc!

I have some questions about CLIP-IQA+:

1. What are the differences between ```clipiqa+``` and ```clipiqa+_rn50_512```?
2. I noticed that the pretrained weights of the ```ctx``` parameters of the ```prompt_learner``` of ```clipiqa+``` have shape (16, 512), while considering ```clipiqa+_rn50_512``` the shape is (2, 16, 512). Could you explain why they differ?
3. I retrained CLIP-IQA+ on FLIVE using the default (except for the positional embedding which I removed) options you provided and the shape of the parameters is (2, 16, 512). I need to load these weights into a CLIP-IQA+ model. Should I use ```clipiqa+``` or ```clipiqa+_rn50_512```?

Thanks in advance.

**Comment by chaofengc** [comment]
Thanks to your question.

I have discussed about this with the author. The original `clipiqa+` only finetune a single prompt, and did not separate them as positive and negative prompts. Although it works well, I personally think that it it not consistent with the format of `clipiqa`. Therefore, I introduce two set trainable prompts, i.e., (2, 16, 512), one for postive and another for negative, just to keep the format consistent. The results are similar. `clipiqa+_rn50_512` is retrained with my setting, while `clipiqa+` is directly taken from original github repo, which is trained with `1024x768` images (the size really matters in the koniq dataset).

**Comment by LorenzoAgnolucci** [comment]
Oh, I see. I did not expect CLIP-IQA+ to use a single prompt, since as far as I know in the paper the authors state that they used two prompts. Anyway, thank you very much, ```clipiqa+_rn50_512``` seems to work well for my purposes.

**Comment by chaofengc** [comment]
To be more clear, the official implementation used one learnable prompt, which are combined with both explicit positive and negative texts, as in the following codes:

https://github.com/chaofengc/IQA-PyTorch/blob/96f24ae7e3f16bd59aece7ea409d5b73949d7f45/pyiqa/archs/clipiqa_arch.py#L66-L67

There are still two prompts during testing.

**Comment by LorenzoAgnolucci** [comment]
Ok now I get it, thanks again!

**Comment by ZachL1** [comment]
Hi, @chaofengc 

I need help with the hyperparameters used to reproduce CLIPIQA+. I initially used the default parameters from the repository (`options/train/CLIPIQA/train_CLIPIQA_koniq10k.yml`) but the training results were not very good. I tried adjusting parameters like `batch size` and `pos_embedding` to match the CLIPIQA paper, but the results were still unsatisfactory. On the contrary, the pre-training weights you provided worked well.

Any tips you can provide would be greatly appreciated!

**Comment by chaofengc** [comment]
Thanks for interest. Here are tips you need to know about CLIPIQA+:
- Performance of CLIPIQA+ cannot compare with SOTA models because it only finetune the visual prompts
- To achieve the best performance, you need to use the larger images $1024\times768$ in koniq10k and modify the data preprocessing part accordingly. The example used small images only for fast testing.

**Comment by ZachL1** [comment]
Thank you very much for your reply!

I apologize for not clarifying the situation in my previous comment. Actually my experiments on koniq10k were successful and CLIPIQA+ achieved the desired performance. 
However, when I attempted to train using SPAQ, the performance was significantly lower. The SRCC of the SPAQ test set was around 0.8, and the SRCC of the zero-shot test on KonIQ10k and LIVEC was less than 0.5. In comparison, when trained on KonIQ10k, the SRCC of the KonIQ10k test set was around 0.9, and the SRCC of the zero-shot test on SPAQ and LIVEC exceeded 0.8.  This doesn't usually happen with the other methods I tested.

Is this normal? Do you have any experience or suggestions for training CLIPIQA+ on SPAQ?
Thanks again!

**Comment by chaofengc** [comment]
From your description, the codes seem to work well.  The problem might be related to the data preprocessing steps.
For example, the image sizes are quite different between SPAQ and other datasets, because images in SPAQ are taken by smart phones and are quite large.

**Comment by ZachL1** [comment]
It was indeed the data preprocessing that was causing the performance degradation, which I eventually solved, thank you very much for your help!
